src/hotspot/share/memory/metaspaceShared.cpp

```diff
@@ -776,6 +777,132 @@ bool MetaspaceShared::try_link_class(InstanceKlass* ik, TRAPS) {
   }
 }
 
+void append_strings(char* buffer, size_t buff_len,  const char* arg) {
+  char* start = buffer + strlen(buffer);
+  snprintf(start, buff_len, "%s ", arg);
+}
+
+void MetaspaceShared::cmd_dump_shared_archive(outputStream* output, const char* cmd, const char* file_name,  TRAPS) {
+  // The existing file will be overwritten.
+  char filename[JVM_MAXPATHLEN];
+  const char* file = file_name;
```

Comment:
Is the variable at line 788 necessary? Could you just pass filename to callees?

Reply:
New solutions using CDS.java to do the dump.